lib/linter/linter.js

```diff
@@ -868,19 +868,6 @@ function resolveParserOptions(parser, providedOptions, enabledEnvironments) {
 		parserOptionsFromEnv,
 		providedOptions || {},
 	);
-	const isModule = mergedParserOptions.sourceType === "module";
-
-	if (isModule) {
-		/*
-		 * can't have global return inside of modules
-		 * TODO: espree validate parserOptions.globalReturn when sourceType is setting to module.(@aladdin-add)
-		 */
-		mergedParserOptions.ecmaFeatures = Object.assign(
-			{},
-			mergedParserOptions.ecmaFeatures,
-			{ globalReturn: false },
-		);
-	}
```

Comment:
We can add here this patch for eslintrc (as it's going to be removed anyway):

```js
// Temporary fix for tests
if (mergedParserOptions.ecmaFeatures?.globalReturn) {
	delete mergedParserOptions.ecmaFeatures.globalReturn;
	mergedParserOptions.sourceType = "commonjs";
}
```

That should fix the failing tests. Then, we could use the branch of https://github.com/eslint/js/pull/692 to test everything else:

```
"espree": "https://gitpkg.vercel.app/Pixel998/js/packages/espree?remove-ecmafeatures-globalreturn&scripts.postinstall=npm%20install%20--ignore-scripts%20%26%26%20npm%20run%20build",
```